projects/packages/publicize/src/social-image-generator/class-rest-token-controller.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Defines the endpoints used for handling tokens for the Social Image Generator.
+ *
+ * @package automattic/jetpack-publicize
+ */
+
+namespace Automattic\Jetpack\Publicize\Social_Image_Generator;
+
+use WP_Error;
+use WP_REST_Controller;
+use WP_REST_Server;
+
+/**
+ * Class used to register token related REST API endpoints used by Social Image Generator.
+ */
+class REST_Token_Controller extends WP_REST_Controller {
+
+	/**
+	 * Register REST API endpoints.
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+		register_rest_route(
+			'jetpack/v4',
+			'/social-image-generator/generate-preview-token',
+			array(
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => array( $this, 'generate_preview_token' ),
+				'permission_callback' => array( $this, 'require_admin_privilege_callback' ),
+				'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+				'schema'              => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Passes the request parameters to the WPCOM endpoint to generate a preview image token.
+	 *
+	 * @param WP_Request $request The request object, which includes the parameters.
+	 * @return array|WP_Error The token or an error.
+	 */
+	public function generate_preview_token( $request ) {
+		$text      = $request->get_param( 'text' );
```

Comment:
Does the presence of the schema here automatically take care of the validations here?

Reply:
Yes that's right it does!